src/main/java/org/prebid/server/cache/CacheService.java

```diff
@@ -326,16 +336,47 @@ private static PutObject createJsonPutObjectOpenrtb(CacheBid cacheBid) {
     /**
      * Makes XML type {@link PutObject} from {@link com.iab.openrtb.response.Bid}. Used for OpenRTB auction request.
      */
-    private static PutObject createXmlPutObjectOpenrtb(CacheBid cacheBid) {
-        if (cacheBid.getBid().getAdm() == null) {
-            return PutObject.of("xml", new TextNode("<VAST version=\"3.0\"><Ad><Wrapper>"
+    private PutObject createXmlPutObjectOpenrtb(CacheBid cacheBid, List<String> videoBidIdsToModify,
+                                                String accountId) {
+        final com.iab.openrtb.response.Bid bid = cacheBid.getBid();
+        String vastXml;
+        if (bid.getAdm() == null) {
+            vastXml = "<VAST version=\"3.0\"><Ad><Wrapper>"
                     + "<AdSystem>prebid.org wrapper</AdSystem>"
-                    + "<VASTAdTagURI><![CDATA[" + cacheBid.getBid().getNurl() + "]]></VASTAdTagURI>"
+                    + "<VASTAdTagURI><![CDATA[" + bid.getNurl() + "]]></VASTAdTagURI>"
                     + "<Impression></Impression><Creatives></Creatives>"
-                    + "</Wrapper></Ad></VAST>"), cacheBid.getTtl());
+                    + "</Wrapper></Ad></VAST>";
         } else {
-            return PutObject.of("xml", new TextNode(cacheBid.getBid().getAdm()), cacheBid.getTtl());
+            vastXml = bid.getAdm();
+        }
+
+        final String bidId = bid.getId();
+        if (CollectionUtils.isNotEmpty(videoBidIdsToModify) && videoBidIdsToModify.contains(bidId)) {
+            vastXml = modifyVastXml(vastXml, bidId, accountId);
+        }
+
+        return PutObject.of("xml", new TextNode(vastXml), cacheBid.getTtl());
+    }
+
+    private String modifyVastXml(String stringValue, String bidId, String accountId) {
+        final String closeTag = "</Impression>";
+        final int closeTagIndex = stringValue.indexOf(closeTag);
+
+        // no impression tag - pass it as it is
+        if (closeTagIndex == -1) {
+            return stringValue;
+        }
+
+        final String impressionUrl = eventsUrlTemplate.replace(BID_ID_PLACEHOLDER, bidId)
```

Comment:
i'd like to see this url template like `/event?t=imp&b=BIDID&f=b&a=ACCOUNT ` combined with `external-url` configuration property.